### PR TITLE
[feat]: Added Open orders page

### PR DESCRIPTION
**Pull request by vpranavkrishna**

### CODE REVIEW

**Comment by siva2204** [comment]
openOrder implementation looks good, one more thing we need add is real-time updates in open orders page

`GetMyOrderUpdates` server side rpc should be used for real-time updates

can be added with globalStreams similar to notfication stream.

with that you have to update/add open orders

File: lib/blocs/portfolio/open_orders/open_orders_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      } else {
+        emit(OpenorderFailure(resp.statusMessage));
+      }
+    } catch (e) {
+      logger.i(e.toString());
+      emit(const OpenorderFailure(failedToReachServer));
+    }
+  }
+
+  Future<void> cancelOpenOrders(int? orderId, bool? isAsk) async {
```
Comment (Sudhindra3000) [inline lib/blocs/portfolio/open_orders/open_orders_cubit.dart:30]:
```suggestion
  Future<void> cancelOpenOrder(int orderId, bool isAsk) async {
```

File: lib/blocs/portfolio/open_orders/open_orders_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import 'package:dalal_street_client/proto_build/actions/GetMyOrders.pb.dart';
+import 'package:equatable/equatable.dart';
+
+part 'open_orders_state.dart';
+
+class OpenOrdersCubit extends Cubit<OpenOrdersState> {
+  OpenOrdersCubit() : super(OpenOrdersInitial());
+  Future<void> getOpenOrders() async {
+    try {
+      final GetMyOpenOrdersResponse resp = await actionClient.getMyOpenOrders(
```
Comment (Sudhindra3000) [inline lib/blocs/portfolio/open_orders/open_orders_cubit.dart:16]:
```suggestion
      final resp = await actionClient.getMyOpenOrders(
```
No need to specify type when declaring using `final` or `var`

Comment (Sudhindra3000) [reply lib/blocs/portfolio/open_orders/open_orders_cubit.dart:16]: Check this in other places too

File: lib/blocs/portfolio/open_orders/open_orders_cubit.dart [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          GetMyOpenOrdersRequest(),
+          options: sessionOptions(getIt()));
+      if (resp.statusCode == GetMyOpenOrdersResponse_StatusCode.OK) {
+        emit(OpenordersSucess(resp));
+      } else {
+        emit(OpenorderFailure(resp.statusMessage));
+      }
+    } catch (e) {
+      logger.i(e.toString());
```
Comment (Sudhindra3000) [inline lib/blocs/portfolio/open_orders/open_orders_cubit.dart:25]:
```suggestion
      logger.e(e);
```
Log exceptions as errors everywhere

File: lib/blocs/portfolio/open_orders/open_orders_state.dart [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+class OpenOrdersInitial extends OpenOrdersState {}
+
+class OpenordersSucess extends OpenOrdersState {
+  final GetMyOpenOrdersResponse res;
+  const OpenordersSucess(this.res);
+  @override
+  List<Object> get props => [res];
+}
+
+class OpenorderFailure extends OpenOrdersState {
```
Comment (Sudhindra3000) [inline lib/blocs/portfolio/open_orders/open_orders_state.dart:19]: Have just one state class `OrderFailure` for this and `CancelorderFailure`. And have a field to specify open or close order.
Could be an enum `OrderType` or a bool `isOpen`

File: lib/blocs/portfolio/open_orders/open_orders_state.dart [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+abstract class OpenOrdersState extends Equatable {
+  const OpenOrdersState();
+
+  @override
+  List<Object> get props => [];
+}
+
+class OpenOrdersInitial extends OpenOrdersState {}
+
+class OpenordersSucess extends OpenOrdersState {
```
Comment (Sudhindra3000) [inline lib/blocs/portfolio/open_orders/open_orders_state.dart:12]:
```suggestion
class GetOpenOrdersSuccess extends OpenOrdersState {
```

File: lib/blocs/portfolio/open_orders/open_orders_state.dart [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+class OpenorderFailure extends OpenOrdersState {
+  final String msg;
+  const OpenorderFailure(this.msg);
+  @override
+  List<Object> get props => [msg];
+}
+
+class CancelorderSucess extends OpenOrdersState {
```
Comment (Sudhindra3000) [inline lib/blocs/portfolio/open_orders/open_orders_state.dart:26]:
```suggestion
class CancelOrderSucess extends OpenOrdersState {
```

Comment (Sudhindra3000) [reply lib/blocs/portfolio/open_orders/open_orders_state.dart:26]: And typo in Success in some class names. Check that too

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

**Comment by Sudhindra3000** [comment]
Seems like you are using an image file for cancel button. There's already a `TertiaryButton` component. Use that instead

**Comment by Sudhindra3000** [comment]
> Looks like you are using an image file for cancel button. There's already a `TertiaryButton` component. Use that instead

Even if there wasn't a widget, you could just create one. Or just inline the required styles. It's not that hard. Don't use images or svg files for simple things like buttons

**Comment by Sudhindra3000** [comment]
> Few other things not related to open orders:
> 
> 1. Rename `NewsDetail` to `NewsDetailPage`, and the file to `news_detail_page.dart`
> 2. `NewsDetail` is accepting these as params:
>    ```dart
>     final String text;
>     final String imagePath;
>     final String headline;
>     final String dur;
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    Just pass a `MarketEvent` object instead. It's the same thing
> 3. There's a field in NewsDetail widget called `String dur`. I'm guessing it's supposed to mean duration. Anyway that field will be removed after above change, but give better names to variables from now.
> 4. You are going from news page to news detail page by doing a manual `Navigator.push(NewsDetail(args))` everywhere. All individual pages must to be declared in `route_generator.dart`. And navigate to them using `Navigator.pushNamed` and pass in the route and required args

Better do these things in a separate pr, after open orders in done

File: lib/navigation/route_generator.dart [hunk 7]
```diff
@@ -79,6 +81,11 @@ class RouteGenerator {
           create: (context) => LoginCubit(context.read()),
           child: LoginPage(),
         );
+      case '/openorders':
```
Comment (dhruv1294) [inline lib/navigation/route_generator.dart:84]: pls name this according to sheet page routes, check dalal_home.dart, do the same for referAndEarn as well

**Review by dhruv1294 — CHANGES_REQUESTED** [review]

**Review by satyamurti — CHANGES_REQUESTED** [review]
remove this image [dalal-street-client](https://github.com/delta/dalal-street-client/tree/openOrders)/[assets](https://github.com/delta/dalal-street-client/tree/openOrders/assets)/[images](https://github.com/delta/dalal-street-client/tree/openOrders/assets/images)/tertiary.png

instead use button widgets defined in `components/`

File: lib/pages/dalal_home/dalal_home.dart [hunk 8]
```diff
@@ -104,9 +104,9 @@ class _DalalHomeState extends State<DalalHome> {
...
             borderRadius: BorderRadius.vertical(top: Radius.circular(10))),
         builder: (_) => DalalHomeBottomSheet(
-          items: _sheetMenu,
-          onItemClick: (index) =>
-              Navigator.of(context).pushNamed(_sheetPageRoutes[index]),
-        ),
+            items: _sheetMenu,
+            onItemClick: (index) {
+              Navigator.of(context).pushNamed(_sheetPageRoutes[index]);
+            }),
```
Comment (Sudhindra3000) [inline lib/pages/dalal_home/dalal_home.dart:110]: Unnecessary changes will cause unnecessary merge conflicts for other working on the same file. Pls undo

File: lib/blocs/open_orders_subscription/openorders_subscription_cubit.dart [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import 'package:bloc/bloc.dart';
```
Comment (Sudhindra3000) [inline lib/blocs/open_orders_subscription/openorders_subscription_cubit.dart:1]: Rename file to `open_orders_sub....`. Do change in other places as well

File: lib/pages/news/newsdetail_page.dart [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import 'package:flutter/material.dart';
 
-import '../theme/colors.dart';
+import '../../theme/colors.dart';
```
Comment (Sudhindra3000) [inline lib/pages/news/newsdetail_page.dart:3]: Rename file to `news_detail_page.dart`. And match the filename with Class name

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

**Review by Sudhindra3000 — APPROVED** [review]

**Review by kelpikz — CHANGES_REQUESTED** [review]
- Make the open orders heading appBar
- Increase top margin for Open orders
- Change color of  Action, Detail, Status to blurred-gray and decrease its font size to mockup
- Make the actions row according to the mock up

- Fix all the overflows ( Make column to ListView or Use SingleChildScrollView )
- Pop up for canceling an order is not working

**Comment by dhruv1294** [comment]
Cancelling orders is working but it would be better if it cancelled order real-time.
So, when user long presses an item, try emitting the state again to make it look real-time

**Comment by dhruv1294** [comment]
Also, remove the tertiary.png since you are not using it currently.

**Pull request closed without merging**